nomad/api/search.py

```diff
@@ -0,0 +1,83 @@
+import nomad.api.exceptions
+
+from nomad.api.base import Requester
+
+
+class Search(Requester):
+    """
+    The endpoint returns matches for a given prefix and context, where a context can be jobs, allocations, evaluations,
+    nodes, deployments, plugins, namespaces, or volumes.
+    When using Nomad Enterprise, the allowed contexts include quotas.
+    Additionally, a prefix can be searched for within every context.
+
+    https://developer.hashicorp.com/nomad/api-docs/search
+    """
+    ENDPOINT = "search"
+
+    def __init__(self, **kwargs):
+        super(Search, self).__init__(**kwargs)
+
+    def __str__(self):
+        return "{0}".format(self.__dict__)
+
+    def __repr__(self):
+        return "{0}".format(self.__dict__)
+
+    def __getattr__(self, item):
+        raise AttributeError
+
+    def search(self, prefix, context):
+        """ The endpoint returns matches for a given prefix and context, where a context can be jobs,
+            allocations, evaluations, nodes, deployments, plugins, namespaces, or volumes.
+
+            https://developer.hashicorp.com/nomad/api-docs/search
+            arguments:
+              - prefix:(str) required, specifies the identifier against which matches will be found.
+                For example, if the given prefix were "a", potential matches might be "abcd", or "aabb".
+              - context:(str) defines the scope in which a search for a prefix operates.
+                Contexts can be: "jobs", "evals", "allocs", "nodes", "deployment", "plugins",
+                "volumes" or "all", where "all" means every context will be searched.
+            returns: dict
+            raises:
+              - nomad.api.exceptions.BaseNomadException
+              - nomad.api.exceptions.URLNotFoundNomadException
+              - nomad.api.exceptions.InvalidParameters
+        """
+        accetaple_contexts = ["jobs", "evals", "allocs", "nodes", "deployment", "plugins", "volumes", "all"]
```

Comment:
nit, could we use tuples instead of lists here? Even with the scope context protecting `acceptable_contexts` it should be immutable.